lib/Models/createCatalogItemFromFileOrUrl.js

```diff
@@ -30,47 +32,65 @@ var createCatalogItemFromFileOrUrl = function(terria, fileOrUrl, dataType, confi
 
     var name = isUrl ? fileOrUrl : fileOrUrl.name;
 
-    var newCatalogItem;
-    if (dataType === 'auto') {
-        newCatalogItem = createCatalogItemFromUrl(name, terria);
-
-        if (newCatalogItem.type === 'ogr' && defaultValue(confirmConversion, true) && !confirm('\
-This file type is not directly supported by '+terria.appName+'.  However, it may be possible to convert it to a known \
-format using the '+terria.appName+' conversion service.  Click OK to upload the file to the '+terria.appName+' conversion service now.  Or, click Cancel \
-and the file will not be uploaded or added to the map.')) {
-            return when.reject('The user declined to use the conversion service.');
-        }
-
-    } else if (dataType === 'other') {
-        if (defaultValue(confirmConversion, true) && !confirm('\
-In order to convert this file to a format supported by '+terria.appName+', it must be uploaded to the '+terria.appName+' conversion service. \
-Click OK to upload the file to the '+terria.appName+' conversion service now.  Or, click Cancel \
-and the file will not be uploaded or added to the map.')) {
-            return when.reject('The user declined to use the conversion service.');
-        }
-
-        newCatalogItem = createCatalogMemberFromType('ogr', terria);
+    if (name.match(/\.(shp|jpg|jpeg|pdf|xlsx|xls|tif|tiff|png|txt|doc|docx)$/)) {
```

Comment:
Would it make sense to flip this logic a bit?  That is, first call `createCatalogItemFromUrl` before checking the extension.  If that fails (i.e. returns `ogr`), _then_ check for some common extensions that definitely won't work with the conversion service before prompting the user to invoke it.

The reason is that the set of supported formats is a plugin point in TerriaJS.  An app can add new ones without changing our library.  And at least one of the types in your reject list (TIFF) is a sensible geospatial format that some apps might actually want to support.  So we should let user-defined handlers take precedence over a hard-coded reject list.